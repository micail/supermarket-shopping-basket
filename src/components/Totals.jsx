import React from 'react';
import { useSelector, useDispatch } from 'react-redux';


const Totals = () => {
  const totals = useSelector((state) => state.totals);
  const { subTotal } = totals;
  const { savings } = totals;
  const savingsLength = savings ? savings.length : 0
  const { totalSavings } = totals;
  const { totalToPay } = totals;

  const printSavings = (savingsArr) => {
    return (
      <>
        <span>Savings</span>
        <br />
        {savingsArr.map((el, index) => {
          const name = el[0];
          const discount = (el[1]).toString();
          return (
            <span key={index} className="savings">
              <span>{name}</span>
              <span>{discount}</span>
              <br />
            </span>
          );
        })
        }
      </>
    );
  };

  return (
    <div className="Totals">
      <span>-----</span>
      <br />
      <span className="label">Sub-total</span>
      <span className="value sub-total">{subTotal}</span>
      <br />
      {savingsLength > 0 ? printSavings(savings) : null}
      <br />
      <span>-----</span>
      <br />
      <span className="label">Total savings</span>
      <span className="value total-savings">{totalSavings}</span>
      <br />
      <span>-------------------------</span>
      <br />
      <span className="label">Total to Pay</span>
      <span className="value total-to-pay">{totalToPay}</span>
    </div>
  );
};

export default Totals;
