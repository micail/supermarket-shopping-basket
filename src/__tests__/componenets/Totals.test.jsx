import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import Totals from '../../components/Totals';

const mockStore = configureMockStore();

describe('<Totals />', () => {
  const store = mockStore({
    priceList: [
      ['Beans', 0.50],
      ['Coke', 0.50],
      ['Oranges', 1.99, 'kg'],
    ],
    receipt: [0, 0, 0, 1, 1, [2, 0.3, 0.6]],
    totals: {
      subTotal: 3.5,
      savings: [['Beans 3 for 2', -0.5], ['Coke 2 for £1', -0.4]],
      totalSavings: -0.9,
      totalToPay: '2.60',
    },
  });
  const wrapper = mount(<Provider store={store}><Totals /></Provider>);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render subTotal', () => {
    expect(wrapper.find('.sub-total').text()).toEqual('3.5');
  });
  it('should render savings', () => {
    expect(wrapper.find('.savings').at(0).text()).toEqual('Beans 3 for 2-0.5');
    expect(wrapper.find('.savings').at(1).text()).toEqual('Coke 2 for £1-0.4');
  });
  it('should render totalSavings', () => {
    expect(wrapper.find('.total-savings').text()).toEqual('-0.9');
  });
  it('should render totalToPay', () => {
    expect(wrapper.find('.total-to-pay').text()).toEqual('2.60');
  });
});
