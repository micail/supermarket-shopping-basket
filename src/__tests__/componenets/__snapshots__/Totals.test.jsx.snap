// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Totals /> should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Totals>
    <ul
      className="Totals col-xs-12 list-group"
    >
      <li
        className="list-group-item border-0 display-4"
      >
        -----
      </li>
      <li
        className="list-group-item border-0 d-flex justify-content-between"
      >
        <span
          className="label h4"
        >
          Sub-total
        </span>
        <span
          className="value sub-total h4"
        >
          3.5
        </span>
      </li>
      <span
        className="list-group-item border-0 h4"
      >
        Savings
      </span>
      <li
        className="list-group-item border-0 d-flex justify-content-between savings"
        key="0"
      >
        <span
          className="label h4"
        >
          Beans 3 for 2
        </span>
        <span
          className="value h4"
        >
          -0.5
        </span>
      </li>
      <li
        className="list-group-item border-0 d-flex justify-content-between savings"
        key="1"
      >
        <span
          className="label h4"
        >
          Coke 2 for Â£1
        </span>
        <span
          className="value h4"
        >
          -0.4
        </span>
      </li>
      <li
        className="list-group-item border-0 display-4"
      >
        -----
      </li>
      <li
        className="list-group-item border-0 d-flex justify-content-between"
      >
        <span
          className="label h4"
        >
          Total savings
        </span>
        <span
          className="value total-savings h4"
        >
          -0.9
        </span>
      </li>
      <li
        className="list-group-item border-0 display-4"
      >
        -------------------------
      </li>
      <li
        className="list-group-item border-0 d-flex justify-content-between"
      >
        <span
          className="label h4"
        >
          Total to Pay
        </span>
        <span
          className="value total-to-pay h4"
        >
          2.60
        </span>
      </li>
    </ul>
  </Totals>
</Provider>
`;
